src/app/clusters/operational-state-server/operational-state-server.h

```diff
@@ -0,0 +1,139 @@
+};
+
+/**
+ * OperationalStateServer is a class to manager the opertional state cluster.
+ * In order to impl class can be used for all operational state aliases so inherit the CommandHandlerInterface
+ * In order to overide the Read interface so inherit the AttributeAccessInterface
+ */
+class OperationalStateServer : public CommandHandlerInterface, public AttributeAccessInterface, public Uncopyable
+{
+
+public:
+    /**
+     * Init the operational state server.
+     * This function must be called after define a OperationalStateServer class object.
+     * @param void
+     * @return CHIP_ERROR CHIP_NO_ERROR on success, or corresponding error code.
+     */
+    CHIP_ERROR Init();
+
+    /**
+     * Shut down the operational state server.
+     * This function must be called before destory a OperationalStateServer class object.
+     * @param void
+     */
+    void Shutdown();
+
+    // Inherited from CommandHandlerInterface
+    void InvokeCommand(HandlerContext & ctx) override;
+
+    /// IM-level implementation of read
+    ///
+    /// Returns appropriately mapped CHIP_ERROR if applicable (may return CHIP_IM_GLOBAL_STATUS errors)
+    CHIP_ERROR Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder) override;
+
+private:
+    EndpointId mEndpointId;
+    ClusterId mClusterId;
+#if 0
+    Delegate * mDelegate;
+    OperationalStateDataProvider mOperationalStateDataProvider;
+    GenericOperationalState mOperationalState;
+    GenericOperationalErrorState mOperationalError;
+#endif
+
+    /**
+     * Handle Command: Pause.
+     */
+    void HandlePauseState(HandlerContext & ctx, const Commands::Pause::DecodableType & req);
+
+    /**
+     * Handle Command: Resume.
+     */
+    void HandleResumeState(HandlerContext & ctx, const Commands::Resume::DecodableType & req);
+
+    /**
+     * Handle Command: Start.
+     */
+    void HandleStartState(HandlerContext & ctx, const Commands::Start::DecodableType & req);
+
+    /**
+     * Handle Command: Stop.
+     */
+    void HandleStopState(HandlerContext & ctx, const Commands::Stop::DecodableType & req);
+
+public:
+    /**
+     * Creates an operational state cluster instance. The Init() function needs to be called for this instance to be registered and
+     * called by the interaction model at the appropriate times.
+     * @param aEndpointId The endpoint on which this cluster exists. This must match the zap configuration.
+     * @param aClusterId The ID of the ModeSelect aliased cluster to be instantiated.
+     * @param aDelegate A pointer to a delegate that will handle application layer logic.
+     */
+    OperationalStateServer(EndpointId aEndpointId, ClusterId aClusterId) :
+        CommandHandlerInterface(MakeOptional(aEndpointId), aClusterId),
+        AttributeAccessInterface(MakeOptional(aEndpointId), aClusterId)
+    {
+
+        mEndpointId = aEndpointId;
+        mClusterId  = aClusterId;
+#if 0
```

Comment:
why this if 0 ? 

Could you not use a magic here ?
And do something like ENABLE_DELEGATE 

futhermore where aDelegate is setup ?

Reply:
It's the code I forget to remove. Remove it.  The delegate is from `Delegate * GetOperationalStateDelegate(EndpointId endpointId, ClusterId clusterId)`,   and setup in examples/all-clusters-app/all-clusters-common/src/operational-state-delegates.cpp